### PR TITLE
Removed Appc from Marathon

### PR DESCRIPTION
**Pull request by kensipe**
Summary:
There is a bunch of code for testing, serialization, conversion and task building... but I have never see it used... it was created by coreos which is now RH.. and as of late 2016 is no longer active https://github.com/appc/spec#-disclaimer-
Confirmed with Jie (author of AppC isolator in Mesos) that AppC is dead to us.

There is debate on this being a breaking change...   While technically that is true from a letter of the law standpoint... there is NO WAY this could be used and we do NOT test it in DCOS.  I don't see how you can break what was not used... I'm a spirit of the law kind of guy

JIRA issues:

### CODE REVIEW

**Comment by timcharper** [comment]
Why don’t we go through the depreciation mechanism for removing this?

**Comment by timcharper** [comment]
I agree with you that it seems rather obvious that we can remove safely this. Our policy for removing features from Marathon has been that we turn off the functionality, by default, for one minor version of Marathon. Usually, this would be:

1) Announce deprecation (1.8.x)
2) Mark the feature as off by default in the next minor release (1.9.x)
3) Disable the feature entirely in next minor release + 1 (1.10.x)

I think given the circumstances, we can probably announce deprecation and disable in the same version (although, I think we did make this decision in the past and got in a little bit of trouble for other things that seemed obvious... I'm thinking it's worth the risk and the damage is small given that if there are users out there that upgrade and depend on it, they'll have a full minor version to migrate off and will be prevented from upgrading again until they fully remove the feature).

Let's add a deprecated feature flag: https://github.com/mesosphere/marathon/blob/master/src/main/scala/mesosphere/marathon/DeprecatedFeatures.scala; softRemove 1.8.0, hardRemove 1.9.0.

**Comment by kensipe** [comment]
https://github.com/mesosphere/marathon/blob/master/src/main/scala/mesosphere/marathon/DeprecatedFeatures.scala#L8 would stop that solution from working me thinks

**Comment by kensipe** [comment]
Hold Merge until 1.9

**Comment by kensipe** [comment]
rebased on master for merge on 1.9

**Comment by jeschkies** [comment]
I like the clean up but should we add a deprecation notice to 1.8? Changing the API like this without any notice up front can break the API for someone.

File: src/main/scala/mesosphere/mesos/TaskGroupBuilder.scala [hunk 1]
```diff
@@ -569,10 +569,6 @@ object TaskGroupBuilder extends StrictLogging {
             docker.setConfig(SecretSerializer.toSecretReference(pullConfig.secret))
           }
           image.setType(mesos.Image.Type.DOCKER).setDocker(docker)
-        case raml.ImageType.Appc =>
-          val appcLabels = (LinuxAmd64 ++ im.labels).toMesosLabels
-          val appc = mesos.Image.Appc.newBuilder.setName(im.id).setLabels(appcLabels)
-          image.setType(mesos.Image.Type.APPC).setAppc(appc)
```
Comment (jeschkies) [inline src/main/scala/mesosphere/mesos/TaskGroupBuilder.scala]: I don't think this is enough. `LinuxAmd64` can go as well.

Comment (kensipe) [reply src/main/scala/mesosphere/mesos/TaskGroupBuilder.scala]: @jeschkies thx!   there was a lot more to remove!   it is done!

**Comment by kensipe** [comment]
https://github.com/mesosphere/marathon/blob/master/changelog.md#appc-is-now-deprecated

**Comment by kensipe** [comment]
@jeschkies https://github.com/mesosphere/marathon/pull/6843/files
deprecation is with this PR.   We didn't do what zen-dog recommended which he is right about.. it would have caused users to be alerted to the pending removal.

**Comment by kensipe** [comment]
merged to latest master

**Pull request merged**